code/game/objects/items/weapons/tools/upgrades.dm

```diff
@@ -0,0 +1,445 @@
+	prefix = "stabilized"
+	precision = 10
+
+/obj/item/weapon/tool_upgrade/refinement/magbit
+	name = "magnetic bit"
+	desc = "Magnetises tools used for handling small objects, reducing instances of dropping screws and bolts."
+	icon_state = "magnetic"
+	required_qualities = list(QUALITY_SCREW_DRIVING, QUALITY_BOLT_TURNING, QUALITY_CLAMPING, QUALITY_BONE_SETTING)
+	prefix = "magnetic"
+	precision = 10
+
+
+/obj/item/weapon/tool_upgrade/refinement/ported_barrel
+	name = "ported barrel"
+	desc = "A barrel extension for a welding tool which helps manage gas pressure and keep the torch steady."
+	icon_state = "ported_barrel"
+	required_qualities = list(QUALITY_WELDING)
+	prefix = "ported"
+	precision = 12
+	bulk_mod = 1
+
+
+
+
+
+
+
+
+
+// 		AUGMENTS: MISCELLANEOUS AND UTILITY
+//------------------------------------------------
+
+//Allows the tool to use a cell one size category larger than it currently uses. Small to medium, medium to large, etc
+/obj/item/weapon/tool_upgrade/augment/cell_mount
+	name = "heavy cell mount"
+	icon_state = "cell_mount"
+	desc = "A bulky adapter which allows oversized power cells to be installed into small tools"
+	req_cell = TRUE
+	prefix = "medium-cell"
+	bulk_mod = 1
+
+/obj/item/weapon/tool_upgrade/augment/cell_mount/can_apply(var/obj/item/weapon/tool/T, var/mob/user)
+	.=..()
+	if (.)
+		if (T.suitable_cell == /obj/item/weapon/cell/medium || T.suitable_cell == /obj/item/weapon/cell/small)
+			if (T.cell)
+				user << SPAN_DANGER("You'll need to remove the power cell before installing this upgrade. It won't be compatible afterwards")
+				return FALSE
+			return TRUE
+		else
+			return FALSE
+
+/obj/item/weapon/tool_upgrade/augment/cell_mount/apply_values()
+	if (!holder)
+		return
+	if (holder.suitable_cell == /obj/item/weapon/cell/medium)
+		holder.suitable_cell = /obj/item/weapon/cell/large
+		prefix = "large-cell"
+	else if (holder.suitable_cell == /obj/item/weapon/cell/small)
+		holder.suitable_cell = /obj/item/weapon/cell/medium
+		prefix = "medium-cell"
+	..()
+
+
+
+
+//Stores moar fuel!
+/obj/item/weapon/tool_upgrade/augment/fuel_tank
+	name = "Expanded fuel tank"
+	desc = "An auxiliary tank which stores 30 extra units of fuel"
+	icon_state = "canister"
+	req_fuel = TRUE
+	prefix = "expanded"
+	bulk_mod = 1
+
+/obj/item/weapon/tool_upgrade/augment/fuel_tank/apply_values()
+	if (..())
+		holder.max_fuel += 30
+
+
+//Penalises the tool, but unlocks several more augment slots.
+/obj/item/weapon/tool_upgrade/augment/expansion
+	name = "expansion port"
+	icon_state = "expand"
+	desc = "A bulky adapter which more modifications to be attached to the tool.  A bit fragile but you can compensate"
+	prefix = "custom"
+	bulk_mod = 2
+	degradation_mult = 1.3
+	precision = -10
+
+/obj/item/weapon/tool_upgrade/augment/expansion/apply_values()
+	if (..())
+		holder.max_upgrades += 3
+
+
+/obj/item/weapon/tool_upgrade/augment/spikes
```

Comment:
This upgrade needs more degradation multiplier, to make tool with this installed less reliable.

In case of damage mods, it should lower workspeed as well.